src/libraries/component-time-scroll-view/TimeScrollViewDimensions.ts

```diff
@@ -0,0 +1,83 @@
+import { useTimeFocus } from 'libraries/context-recording-selection';
+import { Matrix } from 'mathjs';
+import { useMemo } from 'react';
+import { convert1dDataSeries } from 'libraries/util-point-projection';
+import { TimeseriesLayoutOpts } from 'View';
+
+type PartialMargins = {
+    left?: number,
+    right?: number,
+    top?: number,
+    bottom?: number
+}
+
+export type Margins = {
+    left: number,
+    right: number,
+    top: number,
+    bottom: number
+}
+
+
+const defaultMargins: Margins = {
+    left: 30,
+    right: 20,
+    top: 20,
+    bottom: 50
+}
+
+
+export const useDefinedMargins = (margins: Margins | undefined): Margins => {
+    return useMemo(() => margins || defaultMargins, [margins])
+}
+
+
+export const useTimeseriesMargins = (timeseriesLayoutOpts: TimeseriesLayoutOpts | undefined, manualMargins?: PartialMargins | undefined): Margins => {
+    return useMemo(() => {
+        // const {hideTimeAxis, hideToolbar, useYAxis } = timeseriesLayoutOpts || {}
+        const {hideTimeAxis } = timeseriesLayoutOpts || {}
+        // Let's not have this depend on useYAxis for now
+        // because we want all time plots to line up
+        // const yAxisLeftMargin = useYAxis ? 20 : 0
+
+        // Not sure why this should depend on hide Toolbar
+        // const defaultMargins = hideToolbar ? {
+        //             left: 30 + yAxisLeftMargin,
+        //             right: 20,
+        //             top: 20,
+        //             bottom: hideTimeAxis ? 20 : 50
+        //         } : {
+        //             left: 20 + yAxisLeftMargin,
+        //             right: 20,
+        //             top: 10,
+        //             bottom: hideTimeAxis ? 0 : 40
+        //         }
```

Comment:
These have depended on margin because our practice has been to draw axis labels in the margin--the space within the margins should be exactly what's available to the data drawing. So when an axis or toolbar is displayed, we need to pad the margin out a bit more in order to give somewhere for the toolbar contents to be drawn. OTOH, if you're hiding the toolbar, we allocate less space for it.

Reply:
This is something we need to discuss. On the one hand, we want to minimize the margins. On the other hand, we want multiple timeseries plots to line up exactly. For now, I'm just solving the immediate problem. We have to think about what to do.